servers/movie_writer/movie_writer_ogv.cpp

```diff
@@ -0,0 +1,441 @@
+			audio_ch = 4;
+			break;
+		case AudioServer::SPEAKER_SURROUND_51:
+			audio_ch = 6;
+			break;
+		case AudioServer::SPEAKER_SURROUND_71:
+			audio_ch = 8;
+			break;
+	}
+	audio_frames = mix_rate / fps;
+
+	// Set up Ogg output streams.
+	srand(time(NULL));
+	ogg_stream_init(&to, rand()); // Video.
+	ogg_stream_init(&vo, rand()); // Audio.
+
+	// Initialize Vorbis audio encoding.
+	vorbis_info_init(&vi);
+	int ret = 0;
+	if (audio_bitrate == 0) {
+		ret = vorbis_encode_init_vbr(&vi, audio_ch, mix_rate, audio_quality);
+	} else {
+		ret = vorbis_encode_init(&vi, audio_ch, mix_rate, -1, (int)(64870 * (ogg_int64_t)audio_bitrate >> 16), -1);
+	}
+	ERR_FAIL_COND_V_MSG(ret, ERR_UNAVAILABLE, "The Ogg Vorbis encoder couldn't set up a mode according to the requested quality or bitrate.");
+
+	vorbis_comment_init(&vc);
+	vorbis_analysis_init(&vd, &vi);
+	vorbis_block_init(&vd, &vb);
+
+	// Set up Theora encoder.
+	// Theora has a divisible-by-16 restriction for the encoded frame size
+	// scale the picture size up to the nearest /16 and calculate offsets.
+	int pic_w = p_movie_size.width;
+	int pic_h = p_movie_size.height;
+	int frame_w = (pic_w + 15) & ~0xF;
+	int frame_h = (pic_h + 15) & ~0xF;
+	// Force the offsets to be even so that chroma samples line up like we expect.
+	int pic_x = (frame_w - pic_w) / 2 & ~1;
+	int pic_y = (frame_h - pic_h) / 2 & ~1;
+
+	y = (uint8_t *)memalloc(pic_w * pic_h);
+	u = (uint8_t *)memalloc(pic_w * pic_h / 4);
+	v = (uint8_t *)memalloc(pic_w * pic_h / 4);
+
+	th_info_init(&ti);
+	ti.frame_width = frame_w;
+	ti.frame_height = frame_h;
+	ti.pic_width = pic_w;
+	ti.pic_height = pic_h;
+	ti.pic_x = pic_x;
+	ti.pic_y = pic_y;
+	ti.fps_numerator = fps;
+	ti.fps_denominator = 1;
+	ti.aspect_numerator = 1;
+	ti.aspect_denominator = 1;
+	ti.colorspace = TH_CS_UNSPECIFIED;
+	// Account for the Ogg page overhead.
+	// This is 1 byte per 255 for lacing values, plus 26 bytes per 4096 bytes for
+	// the page header, plus approximately 1/2 byte per packet (not accounted for here).
+	ti.target_bitrate = (int)(64870 * (ogg_int64_t)video_bitrate >> 16);
+	ti.quality = video_quality * 63;
+	ti.pixel_fmt = TH_PF_420;
+	td = th_encode_alloc(&ti);
+	th_info_clear(&ti);
+	ERR_FAIL_COND_V_MSG(td == NULL, ERR_UNCONFIGURED, "Couldn't create a Theora encoder instance. Check that the video parameters are valid.");
+
+	// Setting just the granule shift only allows power-of-two keyframe spacing.
+	// Set the actual requested spacing.
+	ret = th_encode_ctl(td, TH_ENCCTL_SET_KEYFRAME_FREQUENCY_FORCE, &keyframe_frequency, sizeof(keyframe_frequency - 1));
+	if (ret < 0) {
+		ERR_PRINT("Couldn't set keyframe interval.");
+	}
+
+	if (vp3_compatible) {
+		ret = th_encode_ctl(td, TH_ENCCTL_SET_VP3_COMPATIBLE, &vp3_compatible, sizeof(vp3_compatible));
+		if (ret < 0) {
+			ERR_PRINT("Could not enable strict VP3 compatibility");
+		}
+	}
+
+	// Reverse the rate control flags to favor a "long time" strategy.
+	if (soft_target) {
```

Comment:
`soft_target` is always `true`, so we can perhaps remove the `if` statement (and keep its contents)s.

Reply:
The library default is false. The documentation doesn't even mention it, but I think it allows the encoder to be a bit loose about the bitrate target.

This is from the source:
```
/**Ignore bitrate buffer underflows.
 * If the encoder uses so many bits that the reservoir of available bits
 *  underflows, ignore the deficit.
 * The encoder will not try to make up these extra bits in future frames.
 * At low rates this may cause the result to be oversized; it should normally
 *  be disabled, which is the default.*/
```

I would remove the related code.